hydra-node/src/Hydra/Ledger/Cardano/Isomorphism.hs

```diff
@@ -81,15 +81,30 @@ fromLedgerData = ScriptDatumForTxIn . fromAlonzoData
 
 -- ** Script
 
-fromPlutusScript :: Plutus.Script -> Script PlutusScriptV1
-fromPlutusScript =
-  PlutusScript PlutusScriptV1 . fromPlutusScript'
+-- TODO: Possibly move upstream?
+class FromPlutusScript (script :: Type -> Type) where
+  fromPlutusScript :: IsPlutusScriptVersion lang => AsType lang -> Plutus.Script -> script lang
+
+instance FromPlutusScript PlutusScript where
+  fromPlutusScript _ script =
+    PlutusScriptSerialised bytes
+   where
+    bytes = toShort . fromLazy . serialise $ script
+
+instance FromPlutusScript Script where
+  fromPlutusScript lang =
+    PlutusScript (plutusScriptVersion lang) . fromPlutusScript lang
```

Comment:
This instance is not required if we change the signature of `mkScriptAddress` to also take a `PlutusScript` (like `mkScriptWitness`). Which will overall allow for a monomorphic `fromPlutusScript` and removes the need of `asScript` as well. Of course the type class and the `IsPlutusScriptVersion` would not be required then as well.. so let's make this simpler like that?

Reply:
Done in #3ae1cc6f645d7225fcaa389bc414c198b1b81653

Note that the `IsPlutusScriptVersion` (renamed as `HasPlutusScriptVersion` actually.. is still necessary unless we force fix the language to Plutus V1, which would still be okay as for now in Hydra, but this little change makes it fairly easy to upgrade later on. I've also good hope that this could be submitted and merged upstream.